saleor/graphql/product/tests/test_variant.py

```diff
@@ -4711,3 +4712,137 @@ def test_product_variant_stocks_delete_mutation_invalid_object_type_of_warehouse
+                    field
+                    code
+                    message
+                }
+            }
+        }
+"""
+
+
+@patch("saleor.plugins.manager.PluginsManager.product_variant_updated")
+def test_product_variant_deactivate_preorder(
+    updated_webhook_mock,
+    staff_api_client,
+    permission_manage_products,
+    preorder_variant_global_and_channel_threshold,
+    preorder_allocation,
+):
+    variant = preorder_variant_global_and_channel_threshold
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.pk)
+
+    allocations_before = Allocation.objects.filter(
+        stock__product_variant_id=variant.pk
+    ).count()
+
+    response = staff_api_client.post_graphql(
+        QUERY_VARIANT_DEACTIVATE_PREORDER,
+        {"id": variant_id},
+        permissions=[permission_manage_products],
+    )
+    variant.refresh_from_db()
+    content = get_graphql_content(response)
+    flush_post_commit_hooks()
+    data = content["data"]["productVariantPreorderDeactivate"]["productVariant"]
+
+    assert data["preorder"]["isPreorder"] is False
+    assert data["stocks"][0]["quantityAllocated"] > allocations_before
+
+    updated_webhook_mock.assert_called_once_with(variant)
+
+
+def test_product_variant_deactivate_preorder_non_preorder_variant(
+    staff_api_client,
+    permission_manage_products,
+    variant,
+):
+    assert variant.is_preorder is False
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.pk)
+
+    response = staff_api_client.post_graphql(
+        QUERY_VARIANT_DEACTIVATE_PREORDER,
+        {"id": variant_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+    error = content["data"]["productVariantPreorderDeactivate"]["errors"][0]
+
+    assert error["field"] == "id"
+    assert error["code"] == ProductErrorCode.INVALID.name
+
+
+@patch("saleor.graphql.product.mutations.products.deactivate_preorder_for_variant")
+def test_product_variant_deactivate_preorder_cannot_deactivate(
+    mock_deactivate_preorder_for_variant,
+    staff_api_client,
+    permission_manage_products,
+    preorder_variant_global_and_channel_threshold,
+    preorder_allocation,
+):
+    variant = preorder_variant_global_and_channel_threshold
+    variant_id = graphene.Node.to_global_id("ProductVariant", variant.pk)
+
+    mock_deactivate_preorder_for_variant.side_effect = PreorderAllocationError(
+        preorder_allocation.order_line
+    )
+
+    response = staff_api_client.post_graphql(
+        QUERY_VARIANT_DEACTIVATE_PREORDER,
+        {"id": variant_id},
+        permissions=[permission_manage_products],
+    )
+    content = get_graphql_content(response)
+    error = content["data"]["productVariantPreorderDeactivate"]["errors"][0]
+
+    assert error["field"] is None
+    assert error["code"] == ProductErrorCode.PREORDER_VARIANT_CANNOT_BE_DEACTIVATED.name
+
+
+def test_product_variant_deactivate_preorder_as_customer(
```

Comment:
Could we add a test for `product_variant_deactivate_preorder` as:
- App with permission
- App without permission